application/controllers/UserManagementController.php

```diff
@@ -420,6 +424,253 @@ public function actionDeleteUser()
+
+        $userId = sanitize_int($userId);
+
+        $aData['userId'] = $userId;
+        $aData['action'] = $action;
+
+        if ($action == 'activate') {
+            $oEvent = new PluginEvent('beforeAdminUserActivation');
+            $oEvent->set('request', App()->request);
+            App()->getPluginManager()->dispatchEvent($oEvent);
+            $isUserLimitReached = $oEvent->get('disable', false);
+
+            $aData['showUpgradeModal'] = (bool)$isUserLimitReached;
+        }
+
+        return $this->renderPartial('partial/confirmuseractivation', $aData);
+    }
+
+
+    /**
+     * Stores the status settings
+     *
+     * @return void|string
+     * @throws CException
+     */
+    public function actionUserActivateDeactivate()
+    {
+        if (!Permission::model()->hasGlobalPermission('users', 'delete')) {
+            return $this->renderPartial(
+                'partial/error',
+                ['errors' => [gT("You do not have permission to access this page.")], 'noButton' => true]
+            );
+        }
+        $userId = sanitize_int(Yii::app()->request->getParam('userid'));
+        $action = Yii::app()->request->getParam('action');
+        $oUser = User::model()->findByPk($userId);
+        if ($oUser) {
+            if ($action == 'activate') {
+                $oEvent = new PluginEvent('beforeAdminUserActivation');
+                $oEvent->set('request', App()->request);
+                App()->getPluginManager()->dispatchEvent($oEvent);
+                $result = $oEvent->get('disable', null);
+                if (!empty($result) && $oEvent->get('disable') == true) {
+                    return $this->renderPartial(
+                        '/admin/super/_renderJson',
+                        [
+                            'data' => [
+                                'success' => false,
+                                'message' => gT('Please upgrade your plan to activate more users.'),
+                                'errors' => gT('Please upgrade your plan to activate more users.')
+                            ]
+                        ]
+                    );
+                }
+            }
+
+            if ($oUser->setActivationStatus($action)) {
+                return App()->getController()->renderPartial('/admin/super/_renderJson', [
+                    'data' => [
+                        'success' => true,
+                        'message' => gT('Status successfully updated')
+                    ]
+                ]);
+            };
+        }
+        return App()->getController()->renderPartial('/admin/super/_renderJson', [
+            'data' => [
+                'success' => false
+            ]
+        ]);
+    }
+
+    /**
+     * Stores the status settings run via MassEdit
+     *
+     * @throws CException
+     */
+    public function actionBatchStatus()
+    {
+        if (!Permission::model()->hasGlobalPermission('users', 'update')) {
+            return $this->renderPartial(
```

Comment:
I think in other places we throw a 403 exception html code.

Reply:
403 is always better for security. Allow to be tracked by other tool.